### ISSUE TITLE
Not able to use custom path

### ISSUE THREAD

**Issue by trygve-lie** [primus/primus#360]
I'm trying to upgrade from latest 2.x version to the 3.x version and I'm having a problem with Primus not accepting a custom path to the socket.

This is my server code (unchanged between the upgrade and works on 2.x):

```js
var socket = new Primus(httpServer, { 
    pathname: '/foo/bar/socket',
    transformer: 'websockets' 
});
```

This is my client code:

```js
var socket  = new Primus('http://localhost:8000/foo/bar/socket?xyz=test');
```

A connection is not established and this is what the browser complains about:

```txt
WebSocket connection to 'ws://localhost:8000/primusexample/?xyz=test' failed: WebSocket is closed before the connection is established.
```

I've kinda tracked down the issue to be on [this line](https://github.com/primus/primus/blob/master/primus.js#L1140) where the `.uri` method is only called with an 'option` Object which never have a `pathname` so the default will always kick in. I think the line should be something like this:

```js
options.pathname = 'pathname' in options ? options.pathname : (url.pathname ? url.pathname : this.pathname.slice(1));
```

**Comment by 3rd-Eden** [comment]
The `new Primus` shouldn't be called with the pathname in the URL it's not something we did support in Primus 2.x if I recall correctly. It's compiled in to the client during the build step as `Primus.prototype.pathname` so it sounds to me that you didn't recompile the client after changing the pathname on the server.

**Comment by trygve-lie** [comment]
The client are recompiled. In my project I've deleted the `node_modules` directory and then done a clean `npm install`. `node_modules\primus\package.json` then show me that I've got version 3.x. I've then manually deleted the client file and then run the compile step (`primus.save(__dirname +'/primus.js');`). A new client file is created.

Though; in my new client file I find: 

```js
Primus.prototype.version = "2.4.12";
```

So; what version do I have? Does the compile step write some temp files or something somewhere?

**Comment by trygve-lie** [comment]
What makes this a bit weird is that the version of 2.x I use is pinned to 2.4.11. Never had version 2.4.12 in.

If I revert my upgrade to 2.4.11 (by deleting node_modules, deleting the client and recompiling a new client file) the client file state:

```js
Primus.prototype.version = "2.4.11";
```

**Comment by 3rd-Eden** [comment]
I think I know what is going on here. The build that is published to npm has a corrupted build of the `/dist/primus.js` library. I'll prepublish in a sec. Not sure how this is even possible.

**Comment by 3rd-Eden** [comment]
Released Primus 3.0.2 with a rebuild client. In order to prevent this from happening in the future I also added `prepublish` script so it's always rebuild. This should resolve the issue. If you still have issues, give me another poke :p

**Comment by trygve-lie** [comment]
Thanks a lot! Much better :clap:
